test/validation/api/scheduler/scheduler.c

```diff
@@ -2348,6 +2347,130 @@ static void scheduler_test_order_wait_1_thread(void)
 	CU_ASSERT(odp_queue_destroy(queue) == 0);
 }
 
+static int order_wait_helper(void *arg ODP_UNUSED)
+{
+	odp_event_t ev;
+
+	ev = odp_schedule(NULL, ODP_SCHED_WAIT);
+	if (ev == ODP_EVENT_INVALID) {
+		odp_atomic_store_rel_u32(&globals->order_wait.helper_ready, 1);
+		return 0;
+	}
+	odp_event_free(ev);
+
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_active, 1);
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_ready, 1);
+
+	/* Wait that the main thread can attempt to overtake us */
+	odp_time_wait_ns(ODP_TIME_SEC_IN_NS);
+
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_active, 0);
+
+	/* We are not interested in further events */
+	odp_schedule_pause();
+	/* Release context */
+	while ((ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT))
+	       != ODP_EVENT_INVALID) {
+		/* We got an event that was meant for the main thread */
+		odp_event_free(ev);
+	}
+
+	return 0;
+}
+
+static void scheduler_test_order_wait(void)
+{
+	odp_schedule_capability_t sched_capa;
+	odp_queue_param_t queue_param;
+	odp_queue_t queue;
+	pthrd_arg thr_arg = {.numthrds = 1};
+	int ret;
+	odp_event_t ev;
+
+	CU_ASSERT(!odp_schedule_capability(&sched_capa));
+
+	sched_queue_param_init(&queue_param);
+	queue_param.sched.sync = ODP_SCHED_SYNC_ORDERED;
+	queue = odp_queue_create("ordered queue", &queue_param);
+	CU_ASSERT_FATAL(queue != ODP_QUEUE_INVALID);
+	CU_ASSERT_FATAL(odp_queue_type(queue) == ODP_QUEUE_TYPE_SCHED);
+	CU_ASSERT_FATAL(odp_queue_sched_type(queue) == ODP_SCHED_SYNC_ORDERED);
+
+	odp_atomic_init_u32(&globals->order_wait.helper_ready, 0);
+	odp_atomic_init_u32(&globals->order_wait.helper_active, 0);
+
+	ret = odp_cunit_thread_create(order_wait_helper, &thr_arg);
+	CU_ASSERT_FATAL(ret == thr_arg.numthrds);
+
+	/* Send an event to the helper thread */
+	enqueue_event(queue);
+
+	/* Wait that the helper thread gets the event */
+	while (!odp_atomic_load_acq_u32(&globals->order_wait.helper_ready))
+		;
```

Comment:
It would be good to have some timeout (e.g. 10 sec) instead of infinitely waiting. Issues are easier to debug in CI when system does not get stuck, but prints some error or asserts.

Reply:
Added a one second timeout in v3